auth/src/main/java/com/redhat/lightblue/rest/auth/CachedRolesProvider.java

```diff
@@ -72,4 +74,9 @@ public CachedRolesProvider(RolesProvider rolesProvider, RolesCache rolesCache) {
         return roles;
     }
 
+    @Override
+    public RolesProviderHealth checkHealth() {
+        return rolesProvider.checkHealth();
+    }
```

Comment:
`If LDAP goes down, we don't necessarily want to bring all lightblue nodes out of rotation`

As you mention, the only roles in the cache are ones that have already been retrieved from LDAP when requests were made (we don't pre-fetch all users/roles in LDAP).  While we don't have a high amount of user/role churn, just because the cache exists does not mean that requests won't start failing if users not in the cache try and make requests.  

I do think we'll need the checks to account for a degraded status at some point as well, in this case, degraded would be when the connection to LDAP was not available, but the roles cache was still serving requests.  As soon as the cache entries expire, a broken connection to LDAP would then change the status to "unhealthy".  Not sure if we know when role cache expiry happens though to do something about it.

Similarly, degraded could be used to communicate when things are returning in their normal timeframes (i.e. LDAP/Mongo requests are slow).  Of course that implies that we have to know how we define slow for whatever operation (which we'll start to learn as we gather more metrics).